crates/nu-cli/src/completions/custom_completions.rs

```diff
@@ -102,17 +103,22 @@ impl<T: Completer> Completer for CustomCompletion<T> {
                         {
                             completion_options.case_sensitive = case_sensitive;
                         }
-                        if let Some(positional) =
-                            options.get("positional").and_then(|val| val.as_bool().ok())
-                        {
-                            completion_options.positional = positional;
-                        }
                         if let Some(algorithm) = options
                             .get("completion_algorithm")
                             .and_then(|option| option.coerce_string().ok())
                             .and_then(|option| option.try_into().ok())
                         {
                             completion_options.match_algorithm = algorithm;
+                            if let Some(positional) =
+                                options.get("positional").and_then(|val| val.as_bool().ok())
+                            {
+                                if !positional
+                                    && completion_options.match_algorithm == MatchAlgorithm::Prefix
+                                {
+                                    log::warn!("Use of the positional option is deprecated. Use the substring match algorithm instead.");
```

Comment:
This `log::warn` call can be moved up to be inside the `if let Some(positional)` block. Even if the `positional` option has no effect, I think it's best for users to know it's deprecated